contracts/factory/APMRegistryFactory.sol

```diff
@@ -108,18 +109,22 @@ contract APMRegistryFactory is APMInternalAppNames {
         // Initialize
         ens.setOwner(node, ensSub);
         ensSub.initialize(ens, node);
-        apm.initialize(ensSub);
 
-        uint16[3] memory firstVersion;
-        firstVersion[0] = 1;
+        if (_initAPM) {
+            // initialize without ruler by default
+            apm.initialize(ensSub);
 
-        acl.createPermission(this, apm, apm.CREATE_REPO_ROLE(), this);
+            uint16[3] memory firstVersion;
+            firstVersion[0] = 1;
 
-        apm.newRepoWithVersion(APM_APP_NAME, _root, firstVersion, registryBase, b("ipfs:apm"));
-        apm.newRepoWithVersion(ENS_SUB_APP_NAME, _root, firstVersion, ensSubdomainRegistrarBase, b("ipfs:enssub"));
-        apm.newRepoWithVersion(REPO_APP_NAME, _root, firstVersion, repoBase, b("ipfs:repo"));
+            acl.createPermission(this, apm, apm.CREATE_REPO_ROLE(), this);
 
-        configureAPMPermissions(acl, apm, _root);
+            apm.newRepoWithVersion(APM_APP_NAME, _root, firstVersion, registryBase, b("ipfs:apm"));
+            apm.newRepoWithVersion(ENS_SUB_APP_NAME, _root, firstVersion, ensSubdomainRegistrarBase, b("ipfs:enssub"));
+            apm.newRepoWithVersion(REPO_APP_NAME, _root, firstVersion, repoBase, b("ipfs:repo"));
+
+            configureAPMPermissions(acl, apm, _root);
+        }
```

Comment:
I don't get why you would like to leave the APM uninitialized. If that's to allow initializing it later using the new syntax implemented above, I'd add a new function here to allow creating a new APM registry using the new initialization logic.